### ISSUE TITLE
Implement upgraded web GUI

### ISSUE THREAD

**Issue by bnvk** [syncthing/syncthing#1381]
As per a chat in IRC with @calmh whereby he showed me some really nice wireframes made by a community member, I'm opening an issue to track this goal. 

UPDATED:
The wireframes exist at https://projects.invisionapp.com/share/AB2A6DYZN#/screens

As I stated in channel, I'm happy to take some swings at the theming and implementation of said new design as it's quite an improvement over the current UX!

**Comment by uok** [comment]
I don't think Syncthing currently needs a new GUI

**Comment by krozycki** [comment]
I must say that the GUI from the link is really nice.

**Comment by dzarda** [comment]
@krozycki Agreed. :+1:
@bnvk It looks amazing. I would be most delighted to have this.

**Comment by jpjp** [comment]
Yeah I don't think syncthing needs a new... oh wow, that's nice :)

**Comment by krozycki** [comment]
@jpjp Exactly! :+1: :D

**Comment by facastagnini** [comment]
That is nice!
Let me know if I can help, I can at least translate into Spanish

**Comment by jkaberg** [comment]
Yeah, very nice! Only missing the sync progress (bar) of a folder (like the current UI has)

**Comment by bnvk** [comment]
@jpjp @krozycki hehe :grin: 

@facastagnini thanks for offering. I'm new to the project and just learning my way around. Hopefully someone can help steer you to the right direction to contribute language translations!

@gillisig really AWESOME work :+1: can't wait to get this implemented :+1:

**Comment by uok** [comment]
you can help translating at https://www.transifex.com/projects/p/syncthing/

**Comment by gillisig** [comment]
Thanks guys! All and any help is welcomed with implementing this design, so step forward if you are interested in helping out. @bnvk and I are planning to start asap :)

**Comment by uok** [comment]
I like your enthusiasm but please provide details on which CSS/HTML framework you are using. Have you thought about themes/skins support? 

I also think you should create a new github project which whould allow to bundle all comments/feedback and keep this issue section here for Syncthing bugs. 

Also how do you plan to beta test the new GUI? You cannot simply rollout such a radical change to 50k+ users :wink: 

User who want to help out can put your stuff into a folder, set the `STGUIASSETS=c:\path\to\gui` variable and then run Syncthing to loads these assets.

**Comment by facastagnini** [comment]
@uok The syncthing Spanish translation was already complete. I helped with the syncthing-android translation.

**Comment by usmanbashir** [comment]
Hey @bnvk & @gillisig, have you guys started work on the new GUI?

I would like to contribute as well. Though, how will we be managing all this?

**Comment by gillisig** [comment]
Hey @usmanbashir, we have both been rather busy the last week so we haven't started. 

I think we will fork Syncthing and work from that until its ready to be merged back. I will let you know when we are getting started. Thanks!

**Comment by usmanbashir** [comment]
@gillisig Thanks, you let me know!

**Comment by Astaoth** [comment]
If it uses less CPU for opening it, it's a good news.

**Comment by lkwg82** [comment]
I really like the wireframes. Wow really fresh impression.

What about some non-technical dashboard? 
I still think, we can attract some btsync users to come to use  free opensource software. And some of those will take part in the community work. Maybe some more shiny designers  and UX people ;)

Cheers

**Comment by majorsl** [comment]
I was trying to start troubleshooting a sync issue today and found that no where in the GUI (that I can find) shows a simple "last synced <time/date>." "Up To Date" is fantastic, but that's only as good as when it thinks it is up to date. Perhaps a little more detail on things like that when you drill down into the share details?

**Comment by AudriusButkevicius** [comment]
Hover over the last syned file, it shows a date.

**Comment by majorsl** [comment]
Ah! Thanks! Except when you're checking on a node that usually has the most current of everything, in which case there is no "Last File Received" file to hover over.  This wouldn't be accurate even if it was there, yes?  The last file received could have been weeks ago when the last file SENT was 30 seconds ago.  I know, I could go to the other node's interface to see the last file received (which in theory would be the last file sent from the other) ...but, well, you can hopefully see where I'm going especially in multi-node setups.

**Comment by AudriusButkevicius** [comment]
Well we can track the time of last data request, but that does not translate to a file, nor it translates to a protocol message saying I am up to date, etc, hence not sure how valuable that is.

If the devices are connected, you can assume it's talking.

**Comment by merlink01** [comment]
+1 for the new Gui, this one is much clearer to understand than the old one.

**Comment by jpjp** [comment]
No milestone for this yet? :(

**Comment by DanTheColoradan** [comment]
Threw down $20 on bounty source - give some dough if you can! Would love to see this implemented.

**Comment by Eddy2909** [comment]
hi guys,

First of all: I'm loving syncthing and will support the project as much as I can by translating and helping out in the forum.
My other love is to layout GUI's :)
So I decided like @bnvk to design a new GUI  hoping to reach the right ones here to support this :)
I already posted my thoughts of a new GUI of syncthing and hope to find some developers, that would attend to developing a new GUI. I know that a new GUI hasn't the pole position of priorities - but maybe there is someone who has the skills to develop this piece of code for the project.
https://forum.syncthing.net/t/my-thoughts-for-a-gui-2-0-looking-for-a-frontend-developer/3002/39

I'm updating the mockups in the forum and my git folder "mockups (https://github.com/Eddy2909/GUI4syncthing) once there are updates

thanks much!

**Comment by gillisig** [comment]
Hey @Eddy2909, it looks like you put a lot of work into that GUI, good job. 

It seems we are at a bit of a crossroad though, since there are two designs now that are each headed in a completely different direction. On one hand we got a design that simplifies the UI, removing a lot of potentially unnecessary info. On the other hand we got a design that adds more bells and whistles giving you more info on everything that is happening. 

If you ask me I don't think Syncthing should be a tool you use to monitor "everything" that is happening between the syncing devices. It should be something you set and forget, it should let you get on with your work knowing your files will be synced. All that info is great when you are working on the product, developing it. But not so much when you just want to get on with your life. 

So I think there needs to be some decision made on where we actually want to take the GUI of Syncthing, just so we don't waste anybodies time and increase efficiency. Probably it would be best if the main contributors @calmh and @AudriusButkevicius would chime in and make a decision on what direction they want to go in.

Personally I don't have much time to spare these days, but I have been putting what little I got into implementing the design I made, for a while now. It shouldn't be too long until its ready for testing.

**Comment by AudriusButkevicius** [comment]
DISCLAIMER: Personal view.

So this is tricky. I guess different users prefer different things hence why this has diverged into two different concepts.

To be honest, I don't care which design wins as long as someone actually puts the effort of implementing in, in order to replace the existing one which obviously doesn't deliver.

Given the feedback we've had from the community, syncthing seems to be complicated (and it is), hence I am personally in favor of @gillisig's design purely on the basis of community feedback of making syncthing grandparent proof.

Though I don't find them too much different, one seems to have more details the other one less.

**Comment by Eddy2909** [comment]
hi @gillisig, hi @bnvk.
indeed we need  a decision which gui will be the default one of syncthing. but since syncthing supports changing the gui, I would love to see both guis developed ;)
But I think we should ask the community! 
I tried to build a constistent, easy to use and clear gui with focus on a good responsive behaviour (for @Nutomic and his app), that grandmoms will find usefull too :)
Since I cancelled the chart above and if you have a detailed look on the guis, I think we're not so far away from eachothers mockups.
For the first throw, I made a doodle (http://doodle.com/p9fyc83qx8qk4p4v) for this - what do you think? @AudriusButkevicius @calmh

**Comment by generalmanager** [comment]
I really like both drafts and would like to see both implemented at some point (not possible in doodle due to radio buttons). But to get to a point where typical Dropbox users enjoy using Syncthing, I'd like to see @bnvk and @gillisig's simplified design as the default and priority regarding the core developers time.

BTW is there any current progress with your design @bnvk @gillisig ?

**Comment by Eddy2909** [comment]
"not possible in doodle due to radio buttons" ?

**Comment by Eddy2909** [comment]
@generalmanager you can now chose both..

**Comment by generalmanager** [comment]
@Eddy2909 Sorry, that was the webdesigner in me. I meant that one can only choose one of the options as you already noticed. In HTML radio buttons are used for forms where only one of the options can be selected at any time. Checkboxes however allow one to select any number of combinations.

**Comment by stooone** [comment]
I think that the "new" GUI looks good, and from new users it will be more simple to start with, and to be honest Syncthing should work out of the box with as little tuning as possible. Also there are many power users who like the possibility to tune things.

I think we can make both groups of users happy, if we go with the simple GUI and have somewhere in the Settings a checkbox what enables all the hidden interface elements to tune all little parts of the system.

**Comment by dreaddy** [comment]
Vote here for the new simplified gui.
I honestly don't see myself spending that much time monitoring activities. 
Not with all the cpu a gui will demand.

That said both designs are amazing. 

The perfect option would be make the gui a plugin of sorts.

**Comment by Eddy2909** [comment]
@dreaddy some months ago, I created a poll for that purpose:

http://doodle.com/p9fyc83qx8qk4p4v

**Comment by AudriusButkevicius** [comment]
It's already pluggable, sadly there is only one implementation to date ;) So less words more actions people.

**Comment by gillisig** [comment]
Just submitted a pull-request here: https://github.com/syncthing/syncthing/pull/2425 its a WIP and you are all welcome to help out!

**Comment by marbens-arch** [comment]
Is this related to the abandoned next-gen-gui experiment?

**Comment by acolomb** [comment]
@marbens-arch I don't think so.  There's references to two distinct new GUI experiments, but neither one resembles the "next-gen-gui" AFAICT.

I don't think there's value in keeping this 10-year-old discussion on the open issues list, as it has clearly failed to deliver.
